src/loki/service.cc

```diff
@@ -162,19 +162,21 @@ namespace valhalla {
 
       // Get the costing options if in the config or make a blank one.
       // Creates the cost in the cost factory
-      std::string method_options = "/costing_options/" + *costing;
-      auto* method_options_ptr = rapidjson::Pointer{method_options}.Get(request);
+      auto* method_options_ptr = rapidjson::Pointer{"/costing_options/" + *costing}.Get(request);
       auto& allocator = request.GetAllocator();
-      if(!method_options_ptr)
-        request.AddMember(rapidjson::Value(method_options, allocator), rapidjson::Value{rapidjson::kObjectType}, allocator);
-      method_options_ptr = rapidjson::Pointer{method_options}.Get(request);
+      if(!method_options_ptr) {
+        auto* costing_options = rapidjson::Pointer{"/costing_options"}.Get(request);
+        if(!costing_options) {
+          request.AddMember(rapidjson::Value("costing_options", allocator), rapidjson::Value(rapidjson::kObjectType), allocator);
+          costing_options = rapidjson::Pointer{"/costing_options"}.Get(request);
+        }
```

Comment:
there could be an `else if(!costing_options->IsObject())` here to detect when someone sends an array instead of an object because rapidjson will throw if they do that. i didnt want increase the pedantry just yet though. i think maybe in the utils header we should add some helper functions to do these detections. for now we catch the exception and report an error (its just not that helpful)